torchgeo/datasets/copernicus/base.py

```diff
@@ -0,0 +1,259 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Copernicus-Bench abstract base class."""
+
+import os
+from abc import ABC, abstractmethod
+from collections.abc import Callable, Sequence
+from typing import Literal
+
+import pandas as pd
+from einops import rearrange
+from matplotlib import pyplot as plt
+from matplotlib.colors import Colormap
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from torchgeo.datasets.geo import NonGeoDataset
+
+from ..errors import DatasetNotFoundError, RGBBandsMissingError
+from ..utils import (
+    Path,
+    download_and_extract_archive,
+    extract_archive,
+    percentile_normalization,
+)
+
+
+class CopernicusBenchBase(NonGeoDataset, ABC):
```

Comment:
This abstract structure seems very general and not specific to CopernicusBench datasets. Are we planning to implement other CopernicusBench tasks?

Reply:
Yes, see Table 3 in https://arxiv.org/abs/2503.11849 for the full list. There are classification, regression, and change detection datasets still coming for LU/LC, flood mapping, and air pollution mapping.